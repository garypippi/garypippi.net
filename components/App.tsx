import { PropsWithChildren } from 'react'
import 'tailwindcss/tailwind.css'

export const App = (props: PropsWithChildren<any>) => {
    return (
        <div className="container mx-auto">
            {props.children}
        </div>
    )
}
