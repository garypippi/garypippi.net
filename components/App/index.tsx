import { PropsWithChildren } from 'react'
import Link from 'next/link'
import { css } from 'goober'
import { color, lg, mergeClass } from '../../modules/css'
import { getEnv } from '../../modules/env'

const { github, twitter } = getEnv()

interface Props {
    className?: string
}

export const App = ({ className, children }: PropsWithChildren<Props>) => {
    return (
        <div className={mergeClass(className, css`
            padding: 40px 0;
        `)}>
            {children}
            <div className={css`
                margin: 80px auto 0;
                padding-top: 10px;
                max-width: ${lg}px;
                border-top: 1px solid ${color['grey-3']};
            `}>
                <Link
                    href="/"
                    className={css`
                        font-size: 16px;
                        color: ${color['grey-1']};
                `}>
                    {'garypippi.net'}
                </Link>
                <span className={css`
                    margin-left: 10px;
                `}>
                    {'|'}
                </span>
                <a
                    href={github}
                    className={css`
                        font-size: 16px;
                        margin-left: 10px;
                        color: ${color['grey-1']};
                `}>
                    {'Github'}
                </a>
                <a
                    href={twitter}
                    className={css`
                        font-size: 16px;
                        margin-left: 10px;
                        color: ${color['grey-1']};
                `}>
                    {'Twitter'}
                </a>
            </div>
        </div>
    )
}
