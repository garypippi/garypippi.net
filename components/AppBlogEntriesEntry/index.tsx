import Link from 'next/link'
import { css } from 'goober'
import { color } from '../../modules/css'
import { Entry } from '../../modules/markdown/types'
import { format } from 'date-fns'

interface Props {
    entry: Omit<Entry, 'body'>
    className?: string
}

export const AppBlogEntriesEntry = ({
    className = '',
    entry: { href, attr },
}: Props) => {
    return (
        <div
            className={
                className +
                ' ' +
                css`
                    display: inline-block;
                    padding: 10px;
                `
            }
        >
            <Link
                href={href}
                className={css`
                    text-decoration: none;
                `}
            >
                <h3
                    className={css`
                        font-weight: normal;
                        font-size: 18px;
                        margin-bottom: 4px;
                        color: ${color['grey-5']};
                    `}
                >
                    {attr.title}
                </h3>
            </Link>
            <span
                className={css`
                    font-size: 16px;
                    color: ${color['grey-1']};
                `}
            >
                {format(new Date(attr.date), 'yyyy/MM/dd HH:mm')}
            </span>
            {attr.tags.map((tag, i) => (
                <Link
                    key={i}
                    href={`/tags/${tag}`}
                    className={css`
                        font-size: 16px;
                        margin-left: 4px;
                        color: ${color['grey-1']};
                    `}
                >
                    {`#${tag}`}
                </Link>
            ))}
        </div>
    )
}
