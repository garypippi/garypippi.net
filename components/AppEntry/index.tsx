import Link from 'next/link'
import { Entry } from '../../modules/markdown/types'
import { css } from 'goober'
import { theme } from '../../modules/csstheme'
import { format } from 'date-fns'

interface Props {
    entry: Entry
}

export const AppEntry = ({ entry: { href, attr } }: Props) => {
    return (
        <Link
            href={href}
            className={css`
                display: block;
                text-decoration: none;
                &:not(:last-child) {
                    margin-bottom: 16px;
                }
        `}>
            <h3 className={css`
                font-size: 20px;
                font-weight: normal;
                margin-bottom: 8px;
                color: ${theme.color.text};
            `}>
                {attr.title}
            </h3>
            <span className={css`
                font-size: 16px;
                color: ${theme.color.text};
            `}>
                {format(new Date(attr.date), 'yyyy/MM/dd HH:mm')}
            </span>
            {attr.tags.map((tag, i) => (
                <span
                    key={i}
                    className={css`
                    font-size: 14px;
                    padding: 2px 4px;
                    margin-left: 4px;
                    color: ${theme.color.text};
                    background-color: ${theme.color.tag};
                `}>
                    {tag}
                </span>
            ))}
        </Link>
    )
}
