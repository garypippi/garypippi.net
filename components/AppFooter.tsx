import { AppIcon } from './AppIcon'
import { useSnsLinks } from '../modules/store'

export const AppFooter = () => {
    return (
        <div className="2xl:px-64 xl:px-48 lg:px-16 md:px-8 px-4 my-8 py-8 text-gray-400">
            <div className="flex flex-row justify-center items-center">
                {useSnsLinks().map(({ href, icon }, i) => href && (
                    <a
                        key={`footer-sns-links-${i}`}
                        href={href}
                        className="mr-2"
                    >
                        <AppIcon
                            size={24}
                            className="inline"
                        >
                            {icon}
                        </AppIcon>
                    </a>
                ))}
            </div>
        </div>
    )
}
