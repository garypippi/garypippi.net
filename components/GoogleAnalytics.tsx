import { NextRouter as Router } from 'next/router'

interface Props {
    id?: string
}

export const idMarker = 'gaid'

export const GoogleAnalytics = (props: Props) => {
    return ! props.id ? (null) : (
        <>
            <script
                className={idMarker}
                type="text/plain"
            >
                {props.id}
            </script>
            <script
                async
                src={`https://www.googletagmanager.com/gtag/js?id=${props.id}`}
            />
            <script
                dangerouslySetInnerHTML={{
                    __html: `
                    window.dataLayer = window.dataLayer || [];
                    function gtag(){dataLayer.push(arguments);}
                    gtag('js', new Date());
                    gtag('config', '${props.id}');
                    `
                }}
            />
        </>
    )
}


export type GtagFn = ((type: string, id: string, params: {page_path: string}) => void)
export type GtagFnOr<T> = GtagFn|T

export declare var gtag: GtagFnOr<undefined>
export const getGtagFn = (): GtagFnOr<false>  => typeof gtag == 'function' && gtag
export const routeEvent = 'routeChangeComplete'

export const createOnRouteChangeComplete = (gtagFn = getGtagFn()) => gtagFn && ((path: string) => {
    const e = document.querySelector(`script.${idMarker}`)
    if (e && e.textContent) {
        gtagFn('config', e.textContent, {
            page_path: path
        })
    }
})

export const useGtag = (router: Router, onRouteChangeComplete = createOnRouteChangeComplete()) => {
    return onRouteChangeComplete && (() => {
        router.events.on(routeEvent, onRouteChangeComplete)
        return () => router.events.off(routeEvent, onRouteChangeComplete)
    })
}
