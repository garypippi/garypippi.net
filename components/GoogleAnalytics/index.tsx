import { useRouter } from 'next/router'
import { useEffect } from 'react'

interface Props {
    id?: string
}

export const idMarker = 'gaid'

export const GoogleAnalytics = (props: Props) => {
    return ! props.id ? (null) : (
        <>
            <script
                className={idMarker}
                type="text/plain"
            >
                {props.id}
            </script>
            <script
                async
                src={`https://www.googletagmanager.com/gtag/js?id=${props.id}`}
            />
            <script
                dangerouslySetInnerHTML={{
                    __html: `
                    window.dataLayer = window.dataLayer || [];
                    function gtag(){dataLayer.push(arguments);}
                    gtag('js', new Date());
                    gtag('config', '${props.id}');
                    `
                }}
            />
        </>
    )
}


export type GtagFn = ((type: string, id: string, params: {page_path: string}) => void)
export const routeEvent = 'routeChangeComplete'

export const createOnRouteChangeComplete = (gid: string, gtag: GtagFn) => {
    return (path: string) => {
        gtag('config', gid, {
            page_path: path
        })
    }
}

export const useGtag = (gtag?: GtagFn|false) => {
    if (! gtag)
        return
    const e = document.querySelector(`script.${idMarker}`)
    if (e && e.textContent) {
        const { events: { on, off } } = useRouter()
        const onRouteChangeComplete = createOnRouteChangeComplete(e.textContent, gtag)
        useEffect(() => {
            on(routeEvent, onRouteChangeComplete)
            return () => {
                off(routeEvent, onRouteChangeComplete)
            }
        })
    }
}
