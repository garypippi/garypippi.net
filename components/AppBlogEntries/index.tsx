import { css } from 'goober'
import { lg, md } from '../../modules/css'
import { Entry } from '../../modules/markdown/types'
import { AppBlogEntriesEntry } from '../AppBlogEntriesEntry'

interface Props {
    entries: Omit<Entry,'body'>[]
}

export const AppBlogEntries = ({ entries }: Props) => {
    return (
        <div className={css`
            max-width: ${lg}px;
            margin: 0 auto;
        `}>
            {entries.map((entry, i) => (
                <AppBlogEntriesEntry
                    key={i}
                    entry={entry}
                    className={css`
                        width: calc(50% - 10px);
                        margin-bottom: 10px;
                        &:not(:nth-of-type(2n)) {
                            margin-right: 10px;
                        }
                `}/>
            ))}
        </div>
    )
}
