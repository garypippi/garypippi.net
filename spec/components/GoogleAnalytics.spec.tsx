import renderer from 'react-test-renderer'
import * as GoogleAnalytics from '../../components/GoogleAnalytics'

describe('GoogleAnalytics.tsx', () => {
    describe('component', () => {
        it('should not render', () => {
            expect(renderer.create(<GoogleAnalytics.GoogleAnalytics />).toJSON()).toBe(null)
        })
        it('should render', () => {
            expect(renderer.create(<GoogleAnalytics.GoogleAnalytics id="hoge" />)).toMatchSnapshot()
        })
    })
    describe('createOnRouteChangeComplete', () => {
        it('should abort', () => {
            expect(GoogleAnalytics.createOnRouteChangeComplete()).toBeFalsy()
        })
        it('should abort further', () => {
            const gtagFn = jest.fn()
            const cb = GoogleAnalytics.createOnRouteChangeComplete(gtagFn)
            expect(cb).toBeTruthy()
            if (cb) {
                cb('hoge')
            }
            expect(gtagFn).not.toHaveBeenCalled()
        })
        it('should run further', () => {
            const gtagFn = jest.fn()
            const cb = GoogleAnalytics.createOnRouteChangeComplete(gtagFn)
            document.head.innerHTML = `<script class="${GoogleAnalytics.idMarker}">hoge</script>`
            expect(cb).toBeTruthy()
            if (cb) {
                cb('fuga')
            }
            expect(gtagFn).toHaveBeenCalledWith('config', 'hoge', {
                page_path: 'fuga'
            })
        })
    })
    describe('useGtag', () => {
        const createRouter = (on?: Function , off?: Function): any => ({
            events: {
                on: on || jest.fn(),
                off: off || jest.fn()
            }
        })
        it('should abort', () => {
            expect(GoogleAnalytics.useGtag(createRouter())).toBeFalsy()
        })
        it('should run', () => {
            const on = jest.fn((_, cb) => cb())
            const off = jest.fn((_, cb) => cb())
            const onRouteChangeComplete = jest.fn()
            const onEffect = GoogleAnalytics.useGtag(createRouter(on, off), onRouteChangeComplete)
            expect(onEffect).toBeTruthy()
            if (onEffect) {
                onEffect()()
                expect(on).toHaveBeenCalledWith(GoogleAnalytics.routeEvent, onRouteChangeComplete)
                expect(off).toHaveBeenCalledWith(GoogleAnalytics.routeEvent, onRouteChangeComplete)
                expect(onRouteChangeComplete).toHaveBeenCalledTimes(2)
            }
        })
    })
})
