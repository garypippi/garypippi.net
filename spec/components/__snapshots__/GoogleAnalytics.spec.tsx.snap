// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleAnalytics.tsx component should render when gid is provided 1`] = `
"<script class=\\"gaid\\" type=\\"text/plain\\">hoge</script><script async=\\"\\" src=\\"https://www.googletagmanager.com/gtag/js?id=hoge\\"></script><script>
                    window.dataLayer = window.dataLayer || [];
                    function gtag(){dataLayer.push(arguments);}
                    gtag('js', new Date());
                    gtag('config', 'hoge');
                    </script>"
`;
